test/test.lua

```diff
@@ -2018,6 +2018,57 @@ function test.multi_gpu_copy_noncontig()
    end
 end
 
+function test.cudaTypeCopy()
+
+   local types = {
+      {'float', 'torch.FloatTensor'},
+      {'byte',  'torch.ByteTensor'},
+      {'char',  'torch.CharTensor'},
+      {'short', 'torch.ShortTensor'},
+      {'long',  'torch.LongTensor'},
+      {'double','torch.DoubleTensor'},
+
+      {'cuda',      'torch.CudaTensor'},
+      {'cudaByte',  'torch.CudaByteTensor'},
+      {'cudaChar',  'torch.CudaCharTensor'},
+      {'cudaShort', 'torch.CudaShortTensor'},
+      {'cudaLong',  'torch.CudaLongTensor'},
+      {'cudaDouble','torch.CudaDoubleTensor'},
+   }
+
+   local N = 100
+   local t0 = torch.range(1,12):reshape(3,4)
+
+   local t = t0:clone()
+   for i = 1, N do
```

Comment:
Because there are n^2 conversions from each type to each other type, so this gets better converage.